data_processing/meteo/pg_processing/task_functions.py

```diff
@@ -0,0 +1,286 @@
+
+
+def delete_and_insert_into_pg(ti):
+    regex_infos = ti.xcom_pull(key="regex_infos", task_ids="download_data")
+    for dataset in DATASETS_TO_PROCESS:
+        tables = []
+        if dataset == "BASE/QUOT":
+            tables.append("base_quot_vent")
+            tables.append("base_quot_autres")
+        else:
+            tables.append(config[dataset]["table_name"])
+        for table in tables:
+            print(table)
+            csv_paths = glob.glob(DATADIR + table + "/*.csv")
+
+            db_params = {
+                'dbname': conn.schema,
+                'user': conn.login,
+                'password': conn.password,
+                'host': conn.host,
+                'port': conn.port,
+            }
+
+            try:
+                conn2 = psycopg2.connect(**db_params)
+                conn2.autocommit = False  # Disable autocommit to start a transaction
+                delete_old_data(conn2, table, regex_infos)
+                for csv_path in csv_paths:
+                    print(csv_path)
+                    load_csv_to_postgres(conn2, csv_path, table, regex_infos)
+
+                # Commit the transaction
+                conn2.commit()
+                print("Transaction committed successfully.")
+
+            except Exception as e:
+                # Rollback the transaction in case of error
+                conn2.rollback()
+                print(f"An error occurred: {e}")
+                print("Transaction rolled back.")
+                raise
+
+            finally:
+                conn2.close()
+
+def load_csv_to_postgres(conn, csv_path, table_name, regex_infos):
+    cursor = conn.cursor()
+    with open(csv_path, 'r') as f:
+        reader = csv.reader(f, delimiter=';')
+        columns = next(reader)  # Read the column names from the first row of the CSV
+        columns.append('DEP')
+        
+        # Temporary file to hold the modified CSV data
+        temp_csv_path = csv_path + '.temp'
+        with open(temp_csv_path, 'w', newline='') as temp_f:
+            writer = csv.writer(temp_f, delimiter=';')
+            writer.writerow(columns)  # Write the column headers to the temp file
+            for row in reader:
+                row = [None if val == '' else val for val in row]
+                row.append(regex_infos[csv_path]["regex_infos"]["DEP"])
+                writer.writerow(row)
+        
+        # Use COPY to load the data into PostgreSQL
+        with open(temp_csv_path, 'r') as temp_f:
+            cursor.copy_expert(f"COPY {SCHEMA_NAME}.{table_name} FROM STDIN WITH CSV HEADER DELIMITER ';'", temp_f)
+    
+    cursor.close()
+    conn.commit()
+    print(f"Data from {csv_path} has been loaded into {table_name}.")
+
+
+
+def delete_old_data(conn, table_name, regex_infos):
+    cursor = conn.cursor()
+    for item in regex_infos:
+        if f"/{table_name}/" in item:
+            query = f"DELETE FROM {SCHEMA_NAME}.{table_name} WHERE 1 = 1 "
```

Comment:
That's a funny workaround, could also be done without `1 = 1` and using `'AND'.join(...)` if you're feeling like changing it